### ISSUE TITLE
Allow users to select a Pokemon box sprite to display in their user card

### ISSUE THREAD

**Issue by scheibo** [Zarel/Pokemon-Showdown-Client#1316]
![example](https://user-images.githubusercontent.com/117249/60751991-26f31b80-9f74-11e9-90fb-5ee984bc5ca2.png)

- box sprites are 40x30, and names can be long enough that they already extend past the end of the usercard (cf. everyone's favorite `'POKEMONDEADCHANNEL'`), so LiD's mock may not work for the final UI
- This personalization feature is **orthogonal to statuses**, and its design/implementation should follow the design of avatars pretty much as closely as possible

**Comment by Zarel** [comment]
Why is it orthogonal to statuses? Statuses seem like the natural place for them.

**Comment by scheibo** [comment]
Do statuses seem like the natural place for avatars as well? These are just a second avatar.

**Comment by Zarel** [comment]
I thought about them as just like emoji in the status line.

If they're avatars, the right place is probably combined with the existing avatar in some way.

**Comment by Zarel** [comment]
I suppose you could also think of a status line as a text-only avatar. That wouldn't be a _bad_ way to think about them. The name, avatar, and status line are pretty related to each other, in that they're mainly places for self-expression, identity, creativity, etc.

**Comment by scheibo** [comment]
I agree, statuses (especially if we had custom presets) could have been implemented in the same way as avatars (and perhaps should have been). The one thing unique about statuses is the 'AFK bit' (`'!'`) which we need in more places to be able to grey the name (ie. with avatars/box sprites and even the status _message_ we can fetch it on demand when the user card is clicked, but we need the away bit to alter sort order/greyness).

I think modeling box sprites after avatars is a better approach because as a general rule, if something is only needed on demand it should only be fetched on demand, as opposed to being sent all the time. Sure, chat bandwidth performance is still dwarfed by asset bandwidth, but why waste resources if we dont need to?

**Comment by Zarel** [comment]
We could roll that back, honestly. Make the AFK bit "does the username end with @" and then remove everything else having to do with status support, replacing it with a system where it accompanies avatars rather than usernames.

There is a sense in which I wanted to display different symbols in the userlist for "Online" vs "Idle" vs "Busy", though.

**Comment by Zarel** [comment]
But I think box sprites should be a part of status so people can choose where in the status to put them in, and also to put multiple in if they want.

**Comment by scheibo** [comment]
> We could roll that back, honestly. Make the AFK bit "does the username end with @" and then remove everything else having to do with status support, replacing it with a system where it accompanies avatars rather than usernames.

I would probably support this.  I would definitely favor this if we go with fixed presets instead of customs. However, I think 'end with @' is perhaps not enough - I think we might want to retain the ability to throw things after the `@` for flexibility (what if we *do* want to treat Busy different than Away or have some new state we want to handle? We wouldnt want to exclude yet another character from usernames, its nicer if we can just keep using `@?` or `@!` etc, no?)

> There is a sense in which I wanted to display different symbols in the userlist for "Online" vs "Idle" vs "Busy", though. But that could also be implemented with symbols only.

Yes, but not only with @. We would need an extra space for this information or we would need to restrict which symbols are in the username further, as mentioned above.

> But I think box sprites should be a part of status so people can choose where in the status to put them in, and also to put multiple in if they want.

These sound like basically emoticons to me. Thank you for explaining this use case, I had no idea what your motivation for including these in status was. The reason I don't see this working in status is that these box sprites are pretty huge - 40x30 and the status line is all of like 10px. I think of these box sprites as a second, smaller avatar (the mock from LiD shows a single sprite, but LiD's mock is flawed in that it misrepresents how big the sprites actually are).

For these to fit into a status I think we'd need to shrink them down to 12x9, at which point I can't see them being distinguishable in any way?

Anyway: if you can include these in statuses somehow I feel like they should also be included anywhere else we can insert text in chat, with some syntax like `[picon:pikachu]`, or ( `!picon:pikachu`, `[picon:25]`,  `!!pikachu!!`  etc, I dont know), but I dont think it really would work given the sizes involved (icon size vs. chat/status font size and line heights). And the reason I'm pushing for them to be treated as a second avatar instead of being part of status is that I don't think they would actually work in this way due to the sizes involved.

**Comment by Zarel** [comment]
> Yes, but not only with @. We would need an extra space for this information or we would need to restrict which symbols are in the username further, as mentioned above.

I like the `@?` and `@!` ideas. It wouldn't require any changes other than continuing to restrict `@`.

> The reason I don't see this working in status is that these box sprites are pretty huge - 40x30 and the status line is all of like 10px.

We could also make status text bigger. Box sprites are big but they don't look _awful_ inlined with text.

**Comment by Zarel** [comment]
https://fontawesome.com/icons/do-not-enter?style=solid

https://fontawesome.com/icons/clock?style=solid

are pretty similar to the Discord icons for "DND" and "Idle" respectively.

**Comment by joebangara43903** [comment]
![image](https://user-images.githubusercontent.com/131228993/232965587-9a66ebe1-e438-4bc6-b6f1-42ab178540c6.png)
